addons/crm/crm_lead.py

```diff
@@ -655,7 +655,7 @@ def merge_opportunity(self, cr, uid, ids, user_id=False, team_id=False, context=
         # An Opportunity always has higher confidence level than a lead, unless its stage probability is 0.0
         for opportunity in opportunities:
             sequence = -1
-            if opportunity.stage_id and not opportunity.stage_id.fold:
```

Comment:
That's not an issue here. The original purpose was to avoid "dummy" stages (previously folded stages, not stages without on_change).